src/config.c

```diff
@@ -164,74 +257,290 @@ static void _Config_SetToDefaults(RedisModuleCtx *ctx) {
-			// User defined size of thread pool.
-			res = _Config_SetThreadCount(ctx, val);
-		} else if(!strcasecmp(param, OMP_THREAD_COUNT)) {
-			// User defined maximum number of OpenMP threads.
-			res = _Config_SetOMPThreadCount(ctx, val);
-		} else if(!strcasecmp(param, VKEY_MAX_ENTITY_COUNT)) {
-			// User defined maximum number of entities per virtual key.
-			res = _Config_SetVirtualKeyEntitiesThreshold(ctx, val);
-		} else if(!strcasecmp(param, MAINTAIN_TRANSPOSED_MATRICES)) {
-			// User specified whether or not to maintain transposed matrices.
-			res = _Config_BuildTransposedMatrices(ctx, val);
-		} else if(!(strcasecmp(param, CACHE_SIZE))) {
-			res = _Config_SetCacheSize(ctx, val);
-		} else {
-			RedisModule_Log(ctx, "warning", "Encountered unknown module argument '%s'", param);
-			return REDISMODULE_ERR;
-		}
+	for(int i = 0; i < argc; i += 2) {
+		// Each configuration is a key-value pair. (K, V)
 
-		// Exit if we encountered an error.
-		if(res != REDISMODULE_OK) return res;
-	}
+		//----------------------------------------------------------------------
+		// get field
+		//----------------------------------------------------------------------
 
-	return res;
-}
+		Config_Option_Field field;
+		RedisModuleString *val = argv[i + 1];
+		const char *field_str = RedisModule_StringPtrLen(argv[i], NULL);
 
-inline int Config_GetThreadCount() {
-	return config.thread_count;
-}
+		// exit if configuration is not aware of field
+		if(!Config_Contains_field(field_str, &field)) {
+			RedisModule_Log(ctx, "warning",
+					"Encountered unknown configuration field '%s'", field_str);
+			return REDISMODULE_ERR;
+		}
 
-inline int Config_GetOMPThreadCount() {
-	return config.omp_thread_count;
-}
+		// exit if encountered an error when setting configuration
+		if(!Config_Option_set(field, val)) {
+			RedisModule_Log(ctx, "warning",
+					"Failed setting field '%s'", field_str);
+			return REDISMODULE_ERR;
+		}
+	}
 
-inline uint64_t Config_GetVirtualKeyEntityCount() {
-	return config.vkey_entity_count;
+	return REDISMODULE_OK;
 }
 
-inline bool Config_MaintainTranspose() {
-	return config.maintain_transposed_matrices;
+bool Config_Option_set(Config_Option_Field field, RedisModuleString *val) {
+	//--------------------------------------------------------------------------
+	// set the option
+	//--------------------------------------------------------------------------
+
+	switch (field)
+	{
+		//----------------------------------------------------------------------
+		// cache size
+		//----------------------------------------------------------------------
+
+		case Config_CACHE_SIZE:
+			{
+				long long cache_size;
+				if(!_Config_ParsePositiveInteger(val, &cache_size)) return false;
+				Config_cache_size_set(cache_size);
+			}
+			break;
+
+			//----------------------------------------------------------------------
+			// OpenMP thread count
+			//----------------------------------------------------------------------
```

Comment:
These comments should be aligned with the `case` statements like `cache size` is.